docs/rfc-1-staking-contract.adoc

```diff
@@ -0,0 +1,381 @@
+misbehavior may result in slashing tokens and thus the entire staked amount is
+indeed at stake.
+
+Beneficiary:: An address where the rewards for participation are sent, earned by
+the operator, on behalf of the owner. A beneficiary doesn’t sign or publish any
+protocol-relevant transactions, but any currency or tokens earned by the
+operator will be transferred to the beneficiary.
+
+Authorizer:: An address appointed by an owner to authorize applications on
+behalf of them. An application must be approved by the authorizer before the
+operator is eligible to participate.
+
+Delegated stake:: An owner’s staked tokens, delegated to the operator by the
+owner. Delegation enables KEEP owners to have their wallets offline and their
+stake operated by operators on their behalf.
+
+== Specification
+
+=== Functionality
+
+==== Delegating a stake
+
+Tokens are delegated by the owner. During the delegation, the owner needs to
+appoint an operator, beneficiary, and authorizer. The owner may decide to
+delegate just a portion of their tokens. The owner may delegate multiple times
+to different operators. For simplicity, the operator address can not be reused
+between delegations even if the delegation has been recovered. The only
+exception is when the delegation has been canceled. To protect against griefing
+by frontrunning stake delegation transactions and reserving operator addresses,
+staking contract should require a governable minimum stake amount for each
+delegation.
+
+Staking contract binds the owner with the operator, beneficiary, and authorizer
+for the given stake delegation and records the stake delegation amount.
+Delegation is set to active after 2 hours, once the cancellation window passes.
+
+Existing KEEP and NU stake owners also need to perform a delegation so that T
+staking contract can record their operator, beneficiary, and authorizer. Since
+these values are read-only, they should be copied from KEEP staking contract to
+T staking contract and NU stakers should set them during the delegation. For
+existing stakes, the entire stake amount is registered in T staking contract.
+
+There are 2 hours to cancel a stake without suffering an undelegation period.
+This functionality is needed e.g. in case of a typo in the operator address.
+Undelegation can not be canceled in the same transaction in which it was
+created. Once the delegation has been canceled, the operator address can be used
+again.
+
+==== Authorizing an application
+
+Before the operator is eligible to participate in the given application, the
+authorizer appointed during the stake delegation needs to review the application
+and approve it to use the stake. From the moment of approval, the application
+is authorized to slash or seize the stake.
+
+The authorizer decides up to which amount the given application is eligible to
+slash. This allows managing the risk profile of the staker individually for each
+application. The authorized stake amount can only be increased.
```

Comment:
Because we notify the application, it would actually be possible to let the authorised stake be reduced if the application approves. This would require some complexity but it would be entirely on the application-side and optional (for the basic behaviour we can just prohibit stake reductions in the application). Probably not worth the complexity though.

Reply:
Yeah... I was worried it would complicate rewards and have some side effects hard to predict right now. I can imagine someone decreasing their authorization just before group selection and increasing it back afterward. They do not take the responsibility but depending on how we implement continuous rewards, they are rewarded for the higher stake amount for most of the time.

I'd leave it as-is just to do not introduce any new complexity if we agree on it.